From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Thu, 23 Apr 2026 16:51:13 +0800
Subject: Set fork as the start method for t_otp.py

As of Python 3.14, the default start method of multiprocessing is
forkserver. However, this introduces an issue since src/tests/t_otp.py
is not defined within a main section, and runs its code during import.
Author: Simon Johnsson <simon.johnsson@canonical.com>
---
 src/tests/t_otp.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/tests/t_otp.py b/src/tests/t_otp.py
index dd5cdc5..51f1ac1 100755
--- a/src/tests/t_otp.py
+++ b/src/tests/t_otp.py
@@ -38,7 +38,8 @@ try:
 except ImportError:
     skip_rest('OTP tests', 'Python pyrad module not found')
 try:
-    from multiprocessing import Process, Queue
+    from multiprocessing import Process, Queue, set_start_method
+    set_start_method('fork', force=True)
 except ImportError:
     skip_rest('OTP tests', 'Python version 2.6 required')
 
